chef/automate
# Verification and Reporting module

### :nut_and_bolt: Description: What code changed, and why?
Changes for Verification and Reporting module

```
chef-automate verify --help
Verify Chef Automate config files and Infrastructure

Usage:
  chef-automate verify [flags]

Flags:
      --certificates                Use this flag to verify the certificates provided in the file
      --file string                 Config file that needs to be verified
      --ha-aws-deploy               Use this flag to verify the AWS Deployment setup and config
      --ha-aws-managed-deploy       Use this flag to verify the AWS Deployment setup and config with Managed Services
      --ha-aws-managed-provision    Use this flag to verify the AWS Provision config with Managed services
      --ha-aws-provision            Use this flag to verify the AWS Provision config
      --ha-onprem-aws-deploy        Use this flag to verify the On-Premise setup and config with AWS Managed services
      --ha-onprem-customer-deploy   Use this flag to verify the On-Premise setup and config with Customer Managed services
      --ha-onprem-deploy            Use this flag to verify the On-Premise setup and config with Chef Managed services
  -h, --help                        help for verify
      --standalone-deploy           Use this flag to verify the Automate standalone setup and config

```


### :chains: Related Resources
Jira Tickets :- https://chefio.atlassian.net/browse/CHEF-1277
https://chefio.atlassian.net/browse/CHEF-835
### :+1: Definition of Done
- Created a Command for verification `chef-automate verify`
- Created a reporting module to print result if chef-automate verify command on terminal in tabular form

### :athletic_shoe: How to Build and Test the Change
- rebuild automate-cli
- start services in hab
- create a dummy OnPrem config for automate,chef-server,postgressql and opensearch ips
- run command `chef-automate verify --ha-onprem-deploy config.toml`

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
[demo-for-verification-and-reporting](https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/EVWSLK6DMxtFsQ3p8GjX8E8BjgdBhEUTVauhKsOFcf70VA?e=lBH6TI)
